clients/src/main/java/org/apache/kafka/common/requests/PushTelemetryRequest.java

```diff
@@ -83,8 +94,38 @@ public Uuid getClientInstanceId() {
         return this.data.clientInstanceId();
     }
 
+    public int getSubscriptionId() {
+        return this.data.subscriptionId();
+    }
+
+    public boolean isClientTerminating() {
+        return this.data.terminating();
+    }
+
+    public String getMetricsContentType() {
+        // Future versions of PushTelemetryRequest and GetTelemetrySubscriptionsRequest may include a content-type
+        // field to allow for updated OTLP format versions (or additional formats), but this field is currently not
+        // included since only one format is specified in the current proposal of the kip-714
+        return null;
+    }
+
+    public ByteBuffer getMetricsData() throws Exception {
+        CompressionType cType = CompressionType.forId(this.data.compressionType());
+        return cType == CompressionType.NONE ?
+                ByteBuffer.wrap(this.data.metrics()) : decompressMetricsData(cType, this.data.metrics());
+    }
+
+    public static ByteBuffer decompressMetricsData(CompressionType compressionType, byte[] metrics) throws Exception {
+        ByteBuffer data = ByteBuffer.wrap(metrics);
+        ByteBuffer decompressedData = ByteBuffer.allocate(10000);
```

Comment:
To avoid too large or too many reallocations I recommend doing something like data.size * 4